poethepoet/plugin.py

```diff
@@ -0,0 +1,77 @@
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from typing import Iterator
+
+import tomlkit
+
+try:
+    # pylint: disable=import-error
+    from cleo.commands.command import Command
+    from poetry.console import application as app
+    from poetry.plugins.application_plugin import ApplicationPlugin
+except ModuleNotFoundError:
+    # Hacky defaults to
+    # tell whether poetry is installed or not,
+    # please mypy,
+    # and please Poetry's plugin system (simulate the ApplicationPlugin object)
+    app = type("application", (), {"Application": None})  # pylint: disable=C0103
+    ApplicationPlugin = type(
+        "ApplicationPlugin",
+        (),
+        {"PLUGIN_API_VERION": "1.0.0", "type": "application.plugin"},
+    )
+
+from . import config
+
+
+def get_poe_task_names() -> Iterator[str]:
+    return (
+        name
+        for name in tomlkit.loads(
+            Path(config.PoeConfig().find_pyproject_toml()).read_text()
+        )["tool"]["poe"]["tasks"].keys()
+        # Make sure it doesn't override default poetry commands
+        if name not in app.COMMANDS
+    )
+
+
+def run_poe_task(name: str, io) -> int:
+    io.fileno = sys.stdin.fileno  # Monkey patches io because subprocess needs it
```

Comment:
Well, I said something [here](https://github.com/nat-n/poethepoet/pull/32#issuecomment-873678025)

> >The functionality you've demonstrated here might be enough for a first release that adds value, but ultimately some thought needs to go into how to best integrate poe CLI functionality with the poetry CLI, for consistently displaying help/errors etc. This is were I imagined most of the work/refactoring will be required, because poe doesn't currently use cleo.
>
> I see. Can't we just redirect subprocess's stdout and stderr some how?